txcache/txListBySenderMap.go

```diff
@@ -7,43 +7,46 @@ import (
 	"github.com/multiversx/mx-chain-storage-go/txcache/maps"
 )
 
-const numberOfScoreChunks = uint32(100)
-
 // txListBySenderMap is a map-like structure for holding and accessing transactions by sender
 type txListBySenderMap struct {
-	backingMap        *maps.BucketSortedMap
+	backingMap        *maps.ConcurrentMap
 	senderConstraints senderConstraints
 	counter           atomic.Counter
-	scoreComputer     scoreComputer
 	txGasHandler      TxGasHandler
-	txFeeHelper       feeHelper
 	mutex             sync.Mutex
 }
 
 // newTxListBySenderMap creates a new instance of TxListBySenderMap
 func newTxListBySenderMap(
 	nChunksHint uint32,
 	senderConstraints senderConstraints,
-	scoreComputer scoreComputer,
 	txGasHandler TxGasHandler,
-	txFeeHelper feeHelper,
 ) *txListBySenderMap {
-	backingMap := maps.NewBucketSortedMap(nChunksHint, numberOfScoreChunks)
+	backingMap := maps.NewConcurrentMap(nChunksHint)
 
 	return &txListBySenderMap{
 		backingMap:        backingMap,
 		senderConstraints: senderConstraints,
-		scoreComputer:     scoreComputer,
 		txGasHandler:      txGasHandler,
-		txFeeHelper:       txFeeHelper,
 	}
 }
 
-// addTx adds a transaction in the map, in the corresponding list (selected by its sender)
-func (txMap *txListBySenderMap) addTx(tx *WrappedTransaction) (bool, [][]byte) {
+// addTxReturnEvicted adds a transaction in the map, in the corresponding list (selected by its sender).
+// This function returns a boolean indicating whether the transaction was added, and a slice of evicted transaction hashes (upon applying sender-level constraints).
+func (txMap *txListBySenderMap) addTxReturnEvicted(tx *WrappedTransaction) (bool, [][]byte) {
 	sender := string(tx.Tx.GetSndAddr())
 	listForSender := txMap.getOrAddListForSender(sender)
-	return listForSender.AddTx(tx, txMap.txGasHandler, txMap.txFeeHelper)
+	tx.precomputeFields(txMap.txGasHandler)
+
+	added, evictedHashes := listForSender.AddTx(tx)
```

Comment:
Upon adding the transaction to the list, the function `listForSender.applySizeConstraints()` is called. For example, if a sender sends one transaction which, by itself alone, breaks the constraints (e.g. transaction is too large), then it will be evicted immediately, and the list would become empty.

However, generally speaking, in practice, that should never be the case, due to:
 - interceptor checks
 - https://github.com/multiversx/mx-chain-mainnet-config/blob/master/config.toml#L391

Thus, removed the lines.